mobile/src/main/java/org/openhab/habdroid/core/OpenHABVoiceService.java

```diff
@@ -112,22 +115,53 @@ private String extractVoiceCommand(Intent data) {
     }
 
 
-    private void sendItemCommand(final String itemName, final String command,
-                                 final Connection conn, final int startId) {
-        Log.d(TAG, "sendItemCommand(): itemName=" + itemName + ", command=" + command);
-        conn.getAsyncHttpClient().post("/rest/items/" + itemName,
-                command, "text/plain;charset=UTF-8", new MyHttpClient.ResponseHandler() {
-                    @Override
-                    public void onSuccess(Call call, int statusCode, Headers headers, byte[] responseBody) {
-                        Log.d(TAG, "Command was sent successfully");
-                        stopSelf(startId);
-                    }
-
-                    @Override
-                    public void onFailure(Call call, int statusCode, Headers headers, byte[] responseBody, Throwable error) {
-                        Log.e(TAG, "Got command error " + statusCode, error);
-                        stopSelf(startId);
-                    }
-                });
+    private void sendVoiceCommand(final Connection conn, final String command, final int startId) {
+        final String commandJson;
+        try {
+            JSONObject commandJsonObject = new JSONObject();
+            commandJsonObject.put("body", command);
+            commandJsonObject.put("Accept-Language", Locale.getDefault().getLanguage());
+            commandJson = commandJsonObject.toString();
+        } catch (JSONException e) {
+            Log.e(TAG, "Could not prepare voice command JSON", e);
+            return;
+        }
+
+        conn.getAsyncHttpClient().post("/voice/interpreters", commandJson,
+                "application/json", new MyAsyncHttpClient.ResponseHandler() {
+            @Override
+            public void onFailure(Call call, int statusCode, Headers headers, byte[] body, Throwable error) {
+                if (statusCode == 404) {
+                    Log.d(TAG, "Voice interpreter endpoint returned 404, falling back to item");
+                    sendRawVoiceCommand(conn, command, startId);
+                } else {
+                    Log.e(TAG, "Sending voice command to new endpoint failed: " + statusCode, error);
+                    stopSelf(startId);
+                }
+            }
+
+            @Override
+            public void onSuccess(Call call, int statusCode, Headers headers, byte[] body) {
+                Log.d(TAG, "Command was sent successfully");
```

Comment:
Add the actual command here?

Reply:
We can do that, but the recognized command is logged in `extractVoiceCommand()` already.